sdcm/utils/common.py

```diff
@@ -1799,3 +1799,12 @@ def has_more_pages(self):
         Returns bool indicating if there are any pages not retrieved.
         """
         return self.future.has_more_pages
+
+
+def str_to_list(str_to_convert, val_type=int):
```

Comment:
why adding this to common.py, I think we have such thing in sct_config.py already...

Reply:
there is a similar one in sct_config.py, but it doesn't make the conversion, only validation. Was this intended so?
```python
def int_or_list(value):
    try:
        value = int(value)
        return value
    except Exception:  # pylint: disable=broad-except
        pass

    if isinstance(value, str):
        try:
            values = value.split()
            [int(v) for v in values]  # pylint: disable=expression-not-assigned
            return value
        except Exception:  # pylint: disable=broad-except
            pass
        try:
            return ast.literal_eval(value)
        except Exception:  # pylint: disable=broad-except
            pass

    raise ValueError("{} isn't int or list".format(value))
```

Reply:
this doesn't only do validation, it do both.

Reply:
```python
In [5]: int_or_list("1 1")                                                      
Out[5]: '1 1'

In [6]: type(int_or_list("1 1"))                                                
Out[6]: str
```
It doesn't convert to list of integers...

Reply:
it's actually historical reasons, since there was a lot of code assume those multi-region parameters are space separated strings.

I complacently forgot about it, but you can still do `int_or_list("1 1").split()` (or go fix all those places that does split, I'll open a task for it)

Reply:
the split will not help, I still need to convert the inner digits to numbers. 
We need to refactor the `int_or_list`, I will create a task for it